### ISSUE TITLE
language variant not working

### ISSUE THREAD

**Issue by seansay** [siznax/wptools#50]
Hi Steve, 

I'm using `zh-cn` to get a wikibase's title, but seems the `label` and `title` is both traditional chinese not simplified chinese(also, what's the difference between these two?) but wikipedia display zh-cn in its page ,

<img width="362" alt="screen shot 2016-11-08 at 3 39 07 pm" src="https://cloud.githubusercontent.com/assets/349342/20090676/971e9c06-a5c9-11e6-9bfa-882fd4f31ac2.png">

```>>> wptools.page(wikibase='Q20474', lang='zh', variant='zh-cn').get(proxy=DOUBAN_PROXY)
Q20474 (zh/zh-cn)
{
  lang: zh
  variant: zh-cn
  wikibase: Q20474
}
www.wikidata.org (wikidata) Q20474
www.wikidata.org (claims) Q1165777|Q8391309|Q188451
zh.wikipedia.org (query) 迴響貝斯
zh.wikipedia.org (parse) 1565553
迴響貝斯 (zh/zh-cn)
{
  cache: <dict(4)> {claims, parse, query, wikidata}
  claims: <dict(3)> {Q1165777, Q188451, Q8391309}
  extext: <str(520)> **回响贝斯**（英语：**Dubstep**），回响贝斯在1990年代诞生，源起英国伦敦南部...
  extract: <str(593)> <p><b>回响贝斯</b>（<span>英语：<span lang="en" xml:la...
  label: 迴響貝斯
  lang: zh
  links: https://en.wikipedia.org/wiki/Dubstep
  modified: <dict(2)> {page, wikidata}
  pageid: 1565553
  parsetree: <str(1268)> <root><template><title>Expand English</titl...
  props: <dict(3)> {P279, P31, P910}
  random: 游铭训
  title: 迴響貝斯
  url: https://zh.wikipedia.org/wiki/%E8%BF%B4%E9%9F%BF%E8%B2%9D%E6%96%AF
  url_raw: https://zh.wikipedia.org/wiki/%E8%BF%B4%E9%9F%BF%E8%B2%9D%E6%96%AF?action=raw
  variant: zh-cn
  wikibase: Q20474
  wikidata: <dict(3)> {category, instance, subclass}
  wikidata_url: https://www.wikidata.org/wiki/Q20474
  wikitext: <str(1000)> {{Expand English|Dubstep}}{{unreferenced|tim...
}
```

**Comment by siznax** [comment]
Hi @seansay!

"label" is the Wikidata title, while "title" is a wikisite title. They only differ slightly in English (titles must begin with a capital letter), but are often the same for Chinese.

```
Wikidata Q20474 "dubstep"
    Traditional Chinese: 迴響貝斯  <= wikidata label
    Chinese: 迴響貝斯              <= wikidata label
zhwiki: 迴響貝斯                   <= zhwiki title
    zh-cn: 回响贝斯                <= zhwiki title (zh-cn variant)
```

**Comment by siznax** [comment]
In your output above, `label` and `title` come from the last request that supports setting those attributes:

[get_wikidata](https://www.wikidata.org/w/api.php?action=wbgetentities&formatversion=2&ids=Q20474&languages=zh-cn&props=info|claims|descriptions|labels|sitelinks&redirects=yes&sites=&titles=&variant=zh-cn) result:

```
"entities": {
    "Q20474": {
        "title": "Q20474",
        "labels": {
            "zh-cn": {
                "language": "zh-cn",
                "value": "迴響貝斯"
            }
        },
        "descriptions": {},
```

[get_query](https://zh.wikipedia.org/w/api.php?action=query&exintro&formatversion=2&inprop=displaytitle|url&list=random&pithumbsize=240&ppprop=wikibase_item&prop=extracts|images|info|pageimages|pageprops|pageterms&redirects&rnlimit=1&rnnamespace=0&titles=%E8%BF%B4%E9%9F%BF%E8%B2%9D%E6%96%AF&variant=zh-cn) result:

```
"query": {
    "pages": [
        {
            "title": "迴響貝斯",
            "extract": "<p><b>回响贝斯</b>..."
...
            "terms": {
                "label": [
                    "迴響貝斯"
```

[get_parse](https://zh.wikipedia.org/w/api.php?action=parse&contentmodel=text&disableeditsection=&disablelimitreport=&disabletoc=&formatversion=2&prop=text|iwlinks|parsetree|wikitext|displaytitle|properties&redirects&page=%E8%BF%B4%E9%9F%BF%E8%B2%9D%E6%96%AF&variant=zh-cn) result:

```
"parse": {
    "title": "迴響貝斯",
```

Unfortunately, we do not have direct access to the `zh-cn` variant title. We also get the `"Unrecognized parameter: 'variant'"` warning, which means we are exploiting an undocumented feature when we ask for the `zh-cn` variant of the `zh.wikipedia.org` page. In other words, we get lucky!

Ideally, the Wikidata "label" and "description" for Chinese (`zh-cn`) would be correct.

**Comment by siznax** [comment]
In any case, we were not getting the label based on `variant='zh-cn'` with `get_wikidata`, so I pushed a fix for that with 89987a4125d8e9ac4418367a61ed863f9f4198d3:

```
>>> p = wptools.page(wikibase='Q20474', lang='zh', variant='zh-cn')
>>> p.get_wikidata()
www.wikidata.org (wikidata) Q20474
www.wikidata.org (claims) Q1165777|Q8391309|Q188451
迴響貝斯 (zh/zh-cn)
{
  label: 迴響貝斯
  lang: zh
  title: 迴響貝斯
  variant: zh-cn
  wikibase: Q20474
  ...
}
```

These language variant issues are complex. We need some solid test cases for them.
